core/src/test/java/org/elasticsearch/index/engine/CombinedDeletionPolicyTests.java

```diff
@@ -79,40 +77,46 @@ public void testKeepCommitsAfterGlobalCheckpoint() throws Exception {
 
         for (int i = 0; i < commitList.size(); i++) {
             if (i < keptIndex) {
-                verify(commitList.get(i), times(1)).delete();
+                assertThat(commitList.get(i).deleteTimes(), equalTo(1));
             } else {
-                verify(commitList.get(i), never()).delete();
+                assertThat(commitList.get(i).deleteTimes(), equalTo(0));
             }
         }
         assertThat(translogPolicy.getMinTranslogGenerationForRecovery(), equalTo(translogGenList.get(keptIndex)));
         assertThat(translogPolicy.getTranslogGenerationOfLastCommit(), equalTo(lastTranslogGen));
     }
 
-    public void testIgnoreSnapshottingCommits() throws Exception {
+    public void testAcquireIndexCommit() throws Exception {
         final AtomicLong globalCheckpoint = new AtomicLong();
         final UUID translogUUID = UUID.randomUUID();
         TranslogDeletionPolicy translogPolicy = createTranslogDeletionPolicy();
         CombinedDeletionPolicy indexPolicy = new CombinedDeletionPolicy(OPEN_INDEX_AND_TRANSLOG, translogPolicy, globalCheckpoint::get);
 
-        long firstMaxSeqNo = randomLongBetween(0, Long.MAX_VALUE - 1);
-        long secondMaxSeqNo = randomLongBetween(firstMaxSeqNo + 1, Long.MAX_VALUE);
+        final long maxSeqNo1 = between(1, 1000);
+        final long translogGen1 = between(1, 100);
+        final MockIndexCommit c1 = mockIndexCommit(maxSeqNo1, translogUUID, translogGen1);
+        final long maxSeqNo2 = maxSeqNo1 + between(1, 1000);
+        final long translogGen2 = translogGen1 + between(1, 100);
+        final MockIndexCommit c2 = mockIndexCommit(maxSeqNo2, translogUUID, translogGen2);
+        globalCheckpoint.set(randomLongBetween(0, maxSeqNo2 - 1)); // Keep both c1 and c2.
+        indexPolicy.onCommit(Arrays.asList(c1, c2));
+        final IndexCommit ref1 = indexPolicy.acquireIndexCommit(true);
+        assertThat(ref1, equalTo(c1));
+        expectThrows(UnsupportedOperationException.class, ref1::delete);
+        final IndexCommit ref2 = indexPolicy.acquireIndexCommit(false);
+        assertThat(ref2, equalTo(c2));
+        expectThrows(UnsupportedOperationException.class, ref2::delete);
+        assertThat(translogPolicy.getMinTranslogGenerationForRecovery(), lessThanOrEqualTo(100L));
```

Comment:
shouldn't this be exactly  translogGen1?

Reply:
lol. I fixed